src/core/thread/mesh_forwarder.cpp

```diff
@@ -1000,52 +940,50 @@ void MeshForwarder::SendEmptyFrame(Mac::Frame &aFrame, bool aAckRequest)
     aFrame.SetFramePending(false);
 }
 
-void MeshForwarder::HandleSentFrame(Mac::Frame &aFrame, otError aError)
+Neighbor *MeshForwarder::UpdateNeighborOnSentFrame(Mac::Frame &aFrame, otError aError, const Mac::Address &aMacDest)
 {
-    Mle::MleRouter &mle = Get<Mle::MleRouter>();
-    Mac::Address    macDest;
-    Neighbor *      neighbor;
-
-    mSendBusy = false;
+    Neighbor *neighbor = NULL;
 
     VerifyOrExit(mEnabled);
 
-    aFrame.GetDstAddr(macDest);
+    neighbor = Get<Mle::MleRouter>().GetNeighbor(aMacDest);
+    VerifyOrExit(neighbor != NULL);
+
+    VerifyOrExit(aFrame.GetAckRequest());
```

Comment:
how about moving this before Line 949?

Reply:
We cannot. Note that this method returns the dest `neighbor` (for use in the rest of code from `HandleSentFrame()`) so we need to get the neighbor independent of (and before we check) whether `aFrame.GetAckRequest()` is set or not.

Reply:
yes, agree that it could not be moved without changing the interface.